modular_skyrat/modules/reagent_forging/code/forge.dm

```diff
@@ -80,213 +139,233 @@
+				return
+			to_chat(user, span_notice("[src] levels up... examine to view what has changed!"))
+			forge_level = FORGE_LEVEL_TWO
+			sinew_lower_chance = 100
+			current_sinew = 10
+			playsound(src, 'sound/magic/demon_consume.ogg', 50, TRUE)
+		if(SKILL_LEVEL_LEGENDARY)
+			if(forge_level >= FORGE_LEVEL_THREE)
+				to_chat(user, span_warning("Your level of smithing has already influenced [src]!"))
+				return
+			to_chat(user, span_notice("[src] levels up... examine to view what has changed!"))
+			forge_level = FORGE_LEVEL_THREE
+			current_core = 3
+			create_reagent_forge()
+			playsound(src, 'sound/magic/demon_consume.ogg', 50, TRUE)
+
 /obj/structure/reagent_forge/attackby(obj/item/I, mob/living/user, params)
 	if(istype(I, /obj/item/stack/sheet/mineral/wood)) //used for weak fuel
+		var/skill_modifier = user.mind.get_skill_modifier(/datum/skill/smithing, SKILL_SPEED_MODIFIER) * 3 SECONDS
 		if(in_use) //only insert one at a time
 			to_chat(user, span_warning("You cannot do multiple things at the same time!"))
 			return
 		in_use = TRUE
-		if(forge_fuel_weak >= 300) //cannot insert too much
-			to_chat(user, span_warning("You only need one to two pieces of wood at a time! You have [forge_fuel_weak] seconds remaining!"))
-			in_use = FALSE
+		if(forge_fuel_weak >= 5 MINUTES) //cannot insert too much
+			fail_message(user, "You only need one to two pieces of wood at a time! You have [forge_fuel_weak/10] seconds remaining!")
 			return
-		to_chat(user, span_warning("You start to throw the fuel into the forge..."))
-		if(!do_after(user, 3 SECONDS, target = src)) //need 3 seconds to fuel the forge
-			to_chat(user, span_warning("You abandon fueling the forge."))
-			in_use = FALSE
+		to_chat(user, span_warning("You start to throw [I] into [src]..."))
+		if(!do_after(user, skill_modifier, target = src)) //need 3 seconds to fuel the forge
+			fail_message(user, "You abandon fueling [src].")
 			return
 		var/obj/item/stack/sheet/stack_sheet = I
 		if(!stack_sheet.use(1)) //you need to be able to use the item, so no glue.
-			to_chat(user, span_warning("You abandon fueling the forge."))
-			in_use = FALSE
+			fail_message(user, "You abandon fueling [src].")
 			return
-		forge_fuel_weak += 300 //5 minutes
+		forge_fuel_weak += 5 MINUTES
 		in_use = FALSE
-		to_chat(user, span_notice("You successfully fuel the forge."))
-		if(prob(30))
-			to_chat(user, span_notice("The forge's fuel lights interestingly..."))
-			addtimer(CALLBACK(src, .proc/spawn_coal), 2 MINUTES)
+		to_chat(user, span_notice("You successfully fuel [src]."))
+		user.mind.adjust_experience(/datum/skill/smithing, 3) //useful fueling means you get some experience
+		if(prob(45))
+			to_chat(user, span_notice("[src]'s fuel lights interestingly..."))
+			addtimer(CALLBACK(src, .proc/spawn_coal), 1 MINUTES)
 		return
 
+	//why use coal over wood? the target temp is set to 100 under coal, while only 50 with wood
```

Comment:
note to self- change target_temp variables to "resting temp" or something similar, just for ease of reading